### PR TITLE
Split out differnt data structures

### PR DESCRIPTION
**Pull request by willsalz**
Also implement naive 'predict probability of a city' function. *Does
not* take into account the fact that the card has already been
reshuffled from an epidemic

cc @anthonybishopric

### CODE REVIEW

**Comment by anthonybishopric** [comment]
I think you meant to open this against anthonybishopric/pandemic-nerd-herd

File: pandemic/record.go [hunk 1]
```diff
@@ -68,3 +71,28 @@ func (c CityDeck) probabilityOfEpidemic() float64 {
...
+
+	// How many cards are left?
+	cardsRemaining := NumInfectionCards - len(gs.InfectionDeck.Drawn)
+
+	// Probability of ANY of the infection cards being the City is equal to 1 minus the probabilty
+	// that *none* of the cards is the city card
+	//
+	// P(C) ~= P(!C)^numCardsRemaining
+	// Assuming 48 cards and infection rate = 4
+	// P(C) = (47/48)*(46/47)*(45/46)*(44/45)
```
Comment (anthonybishopric) [inline pandemic/record.go]: Not sure this is right. Looking at that product, it all cancels to 44/48. If you had an infect rate of 3, it'd be 45/48, which is higher.

Comment (anthonybishopric) [inline pandemic/record.go]: Actually yeah, 1- that value seems fine. Nevermind

**Comment by willsalz** [comment]
Yeah I put the diff up against my other diff so it wouldn't look huge…that works in Phabricator#Differential but I wasn't sure how it would work in GitHub. I guess I should have known.

**Comment by willsalz** [comment]
whoops PR'd against wrong repo

**Pull request closed without merging**